lib/block_view_validator.go

```diff
@@ -1937,6 +1937,65 @@ func (bav *UtxoView) GetGlobalActiveStakeAmountNanos() (*uint256.Int, error) {
 	return globalActiveStakeAmountNanos, nil
 }
 
+func (bav *UtxoView) JailAllInactiveValidators(blockHeight uint64) error {
+	// First, iterate through all of the !isDeleted ValidatorEntries in the UtxoView and
+	// jail any that are inactive.
+	var utxoViewValidatorPKIDs []*PKID
+	for _, validatorEntry := range bav.ValidatorPKIDToValidatorEntry {
+		if validatorEntry.isDeleted {
+			continue
+		}
+
+		// Check if we should jail the validator.
+		shouldJailValidator, err := bav.ShouldJailValidator(validatorEntry, blockHeight)
+		if err != nil {
+			return errors.Wrapf(
+				err,
+				"JailAllInactiveValidators: problem determining if should jail validator %v: ",
+				validatorEntry.ValidatorPKID,
+			)
+		}
+
+		// Jail them if so.
+		if shouldJailValidator {
+			if err = bav.JailValidator(validatorEntry); err != nil {
+				return errors.Wrapf(
+					err, "JailAllInactiveValidators: problem jailing validator %v: ", validatorEntry.ValidatorPKID,
+				)
+			}
+		}
+
+		// We don't want to retrieve any ValidatorEntries from the db that are present in the UtxoView.
+		utxoViewValidatorPKIDs = append(utxoViewValidatorPKIDs, validatorEntry.ValidatorPKID)
+	}
+
+	// Second, iterate through all the ValidatorEntries in the db and jail any that are inactive.
+	dbValidatorEntries, err := DBEnumerateAllCurrentValidators(bav.Handle, utxoViewValidatorPKIDs)
+	if err != nil {
+		return errors.Wrapf(err, "JailAllInactiveValidators: problem retrieving ValidatorEntries: ")
+	}
+
+	for _, validatorEntry := range dbValidatorEntries {
+		// Check if we should jail the validator.
+		shouldJailValidator, err := bav.ShouldJailValidator(validatorEntry, blockHeight)
+		if err != nil {
+			return errors.Wrapf(
+				err, "JailAllInactiveValidators: problem determining if should jail validator %v: ", validatorEntry.ValidatorPKID,
+			)
+		}
+		// Jail them if so.
+		if shouldJailValidator {
+			if err = bav.JailValidator(validatorEntry); err != nil {
```

Comment:
reduce nested if? up to you really, but I prefer to `continue` instead of the nested if err pattern

Reply:
Agreed. Good suggestion